dev-haskell/active/active-0.2.0.14-r1.ebuild

```diff
@@ -0,0 +1,44 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+# ebuild generated by hackport 0.6.7.9999
+
+CABAL_FEATURES="lib profile haddock hoogle hscolour test-suite"
+inherit haskell-cabal
+
+DESCRIPTION="Abstractions for animation"
+HOMEPAGE="https://hackage.haskell.org/package/active"
+SRC_URI="
+	https://hackage.haskell.org/package/${P}/${P}.tar.gz
+	https://hackage.haskell.org/package/${P}/revision/1.cabal -> ${PF}.cabal
+	"
+
+LICENSE="BSD"
+SLOT="0/${PV}"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
+
+RDEPEND=">=dev-haskell/lens-4.0:=[profile?] <dev-haskell/lens-4.20:=[profile?]
+	>=dev-haskell/linear-1.14:=[profile?] <dev-haskell/linear-1.22:=[profile?]
+	>=dev-haskell/semigroupoids-1.2:=[profile?] <dev-haskell/semigroupoids-5.4:=[profile?]
+	>=dev-haskell/semigroups-0.1:=[profile?] <dev-haskell/semigroups-0.20:=[profile?]
+	>=dev-haskell/vector-0.10:=[profile?]
+	>=dev-lang/ghc-7.4.1:=
+"
+DEPEND="${RDEPEND}
+	>=dev-haskell/cabal-1.18
+	test? ( >=dev-haskell/quickcheck-2.9 <dev-haskell/quickcheck-2.14 )
```

Comment:
Should this upper bound on quickcheck be removed?

Reply:
So there actually hasn't been a lot of discussion around this, just sort of me cavalierly Doing Thing Different™

Let me explain my method first, and then my motivation after. I am definitely open to comments/opinions.

1. If newer revision is available on hackage, re-run hackport. This creates the new upperbounds that you see above
2. Since there is a new revision, we need to pull the new cabal file from upstream. That's what [this line](https://github.com/gentoo-haskell/gentoo-haskell/pull/1118/files/8de67feaa5eeeb8dd547c21007be98ef89a89880#diff-2a249ad53cc3b6c77458bf986101bf149e1641ba90b5e435a890d37eff76aa6aR15) does, as well as [this line](https://github.com/gentoo-haskell/gentoo-haskell/pull/1118/files/8de67feaa5eeeb8dd547c21007be98ef89a89880#diff-2a249ad53cc3b6c77458bf986101bf149e1641ba90b5e435a890d37eff76aa6aR39)
2. If the `RDEPEND` has changed, rename the fise `git mv ${P}.ebuild ${P}-r1.ebuild` or some such
3. Check remaining upperbounds - are any of these still old/stale?
4. If yes, manually change them using `cabal_chdeps`, remove the upperbound entirely (as you've suggested here), and file an upstream issue.

So, as to the motivation, it goes something like this - I think that upstream changes should be preferred to anything we do locally in gentoo-haskell. This is why the first step is to pull the new revision and cabal file from upstream.

To directly answer your question though: I prefer to keep upstream's upperbound in tact, so that when grepping the tree I can easily tell what's what - if I see an old upperbound, but the same ebuild is already pulling a revision from upstream, then this indicates to me that there may be a newer revision, which should be a pretty straightforward fix/update. 

If I were to take out the upperbound altogether from the ebuild, then some enterprising young gentooer may copy/paste the old ebuild to pull a new version but not realize that they either need to (1) pull a revised cabal from upstream, or (2) bump the version in the current cabal file using `cabal_chdeps`.

I'm not saying this is right or perfect, just my motivation and thought process. Again, I am open to comments/criticisms.

Reply:
@trofi unless you're against this style of updating dependencies, I'll mark this request as "resolved" and continue to use it

Reply:
I think that is a reasonable approach.